File: SDKs/ticketmaster.com/discovery/v2/typescript/src/sdk/models/operations/getimages.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class GetImagesPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: string;
}

export enum GetImagesIncludeLicensedContentEnum {
    Yes = "yes",
    No = " no"
}


export class GetImagesQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=includeLicensedContent" })
  includeLicensedContent?: GetImagesIncludeLicensedContentEnum;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=locale" })
  locale?: string;
}


export class GetImagesRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetImagesPathParams;

  @SpeakeasyMetadata()
  queryParams: GetImagesQueryParams;
}


export class GetImagesResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  body?: Uint8Array;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: SDKs/ticketmaster.com/discovery/v2/python/src/sdk/models/operations/getimages.py
import dataclasses
from typing import Optional
from enum import Enum


@dataclasses.dataclass
class GetImagesPathParams:
    id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'id', 'style': 'simple', 'explode': False }})
    
class GetImagesIncludeLicensedContentEnum(str, Enum):
    YES = "yes"
    NO = " no"


@dataclasses.dataclass
class GetImagesQueryParams:
    include_licensed_content: Optional[GetImagesIncludeLicensedContentEnum] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'includeLicensedContent', 'style': 'form', 'explode': True }})
    locale: Optional[str] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'locale', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class GetImagesRequest:
    path_params: GetImagesPathParams = dataclasses.field()
    query_params: GetImagesQueryParams = dataclasses.field()
    

@dataclasses.dataclass
class GetImagesResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    body: Optional[bytes] = dataclasses.field(default=None)
    
